### PR TITLE
Create a button project on samd21 with mbed sdk

### PR DESCRIPTION
**Pull request by Salem-Luos**
# PR Checklist 

### _1- PR Comments_
    *If needed, to be filled by developer*  

<br><hr/>
*WARNING : Do not edit the checklist below.*  
<hr/><br>

### _2- Developer section_

- [x] [Documentation] is up to date with new feature
- [x] [Tests] are *passed OK* (Non regression, new features & bug fixes)
- [x] [Code Quality] Remind to check if :
    * Each function has a header (description, inputs, outputs) 
    * Code is commented (particularly in hard to understand areas)
    * There are no new warnings that can be corrected
    * Commits policy is respected (constitancy commits, clear commits comments)
<hr/><br>

### _3- QA section_

- [x] [Review] tests for new features have been *reviewed*
- [x] [Changelog] is up-to-date with expected tags  
    🆕 Feature: [Feature] Description...  
    🆕 Added: [Feature] Description...  
    🆕 Changed: [Feature] Description...  
    🛠️ Fix: [Feature] Description...

### CODE REVIEW

File: examples/projects/SAMD21XPLAINED/button/lib/README [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (nicolas-rabault) [inline examples/projects/SAMD21XPLAINED/button/lib/README:1]: You should remove this file.

File: examples/projects/SAMD21XPLAINED/button/lib/button/button.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+{
+    if (msg->header.cmd == GET_CMD)
+    {
+        // fill the message infos
+        msg_t pub_msg;
+        pub_msg.header.cmd         = IO_STATE;
+        pub_msg.header.target_mode = ID;
+        pub_msg.header.target      = msg->header.source;
+        pub_msg.header.size        = sizeof(char);
+        pub_msg.data[0]            = true; // BTN VALUE
```
Comment (nicolas-rabault) [inline examples/projects/SAMD21XPLAINED/button/lib/button/button.c:55]: You should get a real BTN value.

**Review by nicolas-rabault — CHANGES_REQUESTED** [review]
This button doesn't read any pin!

**Review by JeromeGalan — APPROVED** [review]

**Review by nicolas-rabault — APPROVED** [review]
ok

**Pull request merged**